File: src/com/brianc/css/LengthValue.java
package com.brianc.css;

public class LengthValue implements Value {
	float length;
	Unit unit;

	public LengthValue(float length, Unit unit) {
		super();
		this.length = length;
		this.unit = unit;
	}
	
	public float toPx() {
		return length;
	}
	
	public Value clone() {
		return new LengthValue(length, unit);
	}
	
	@Override
	public String toString() {
		return String.format("LengthValue(%f, %s)", length, unit);
	}
	
	@Override
	public boolean equals(Object obj) {
		if (obj instanceof LengthValue) {
			LengthValue asLength = (LengthValue)obj;
			
			return asLength.length == length && asLength.unit == unit;
		}
		
		return false;
	}
}

File: src/com/brianc/css/LengthValue.kt
package com.brianc.css

class LengthValue(internal var length: Float, internal var unit: Unit) : Value {

    override fun toPx(): Float {
        return length
    }

    override fun clone(): Value {
        return LengthValue(length, unit)
    }

    override fun toString(): String {
        return String.format("LengthValue(%f, %s)", length, unit)
    }

    override fun equals(obj: Any?): Boolean {
        if (obj is LengthValue) {

            return obj.length == length && obj.unit == unit
        }

        return false
    }
}
